kinode/packages/kns-indexer/kns-indexer/src/lib.rs

```diff
@@ -254,22 +361,19 @@ fn main(our: Address, mut state: State) -> anyhow::Result<()> {
                     // TODO: make sure we've seen the whole block, while actually
                     // sending a response to the proper place.
                     Response::new()
-                        .body(IndexerResponse::Name(state.names.get(hash).cloned()))
+                        .body(IndexerResponse::Name(state.get_name(hash)))
                         .send()?;
                 }
 
                 IndexerRequest::NodeInfo(NodeInfoRequest { ref name, .. }) => {
```

Comment:
note: should we add a kimap.get() call here (if we happened to miss an event), or have a separate IndexerRequest for it?

Reply:
yes we should. lazy eval/correctness in indexer. also in `net` runtime module need to do same thing.